File: backend-model/src/main/java/outcobra/server/model/dto/UserDto.java
package outcobra.server.model.dto;

import outcobra.server.model.User;
import outcobra.server.model.interfaces.*;

import static outcobra.server.model.interfaces.ParentLink.make;

public class UserDto implements Identifiable, ParentLinkedDto {
    private final Long id;
    private final String userId, username;

    public UserDto() {
        this.id = null;
        this.userId = this.username = null;
    }

    public UserDto(Long id, String userId, String username) {
        this.id = id;
        this.userId = userId;
        this.username = username;
    }

    public String getUserId() {
        return userId;
    }

    public String getUsername() {
        return username;
    }

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public ParentLink getParentLink() {
        return make(id, User.class);
    }
}

File: backend/src/main/kotlin/outcobra/server/model/dto/UserDto.kt
package outcobra.server.model.dto

import outcobra.server.model.User
import outcobra.server.model.interfaces.OutcobraDto
import outcobra.server.model.interfaces.ParentLink

data class UserDto(val id: Long = 0, val username: String = "", val mail: String = "") : OutcobraDto {
    override val identifier: Long
        get() = id

    override val parentLink: ParentLink
        get() = ParentLink.make(id, User::class.java)
}